File: work/P201.cpp
#include <iostream>
#include <cmath>
using namespace std; 

int main(){
    int inset[100];
    int maxsum = 0;
    for (int i = 1; i <= 100; i++){
        inset[i-1] = pow(i, 2);
        maxsum += pow(i, 2);
    }

    int coeffs[51][maxsum];

    for (int val : inset){
        cout << pow(val, 0.5);
        for (int setSize = 49; setSize > -1; setSize--){
            for (int idx = 0; idx < maxsum; idx++){
                coeffs[setSize+1][idx+1] += coeffs[setSize][idx];
            }
        }
    }
    
    int rv;
    for (int i = 0; i < maxsum; i++){
        if (coeffs[50][i] == 1){
            rv += i;
        }
    }
    cout << rv;
    return 0;
}

File: work/P201.py
from copy import deepcopy as dc
import math
inset = [x**2 for x in range(1,101)]
maxnum = sum(inset[50:]) + 1
print(maxnum)
l = [0]*maxnum
coeffs = []
for i in range(51): coeffs.append(dc(l))
coeffs[0][0] = 1
for i in inset:
    print(math.sqrt(i))
    for setsize in range(len(coeffs)-2,-1,-1):
        for index in range(len(coeffs[setsize])):
            if coeffs[setsize][index] != 0:
                coeffs[setsize+1][index+i] += coeffs[setsize][index]
s = 0
for i in range(len(coeffs[50])): 
    if coeffs[50][i] == 1: s += i
print(s)